api/v1beta1/solrcloud_types.go

```diff
@@ -971,3 +996,47 @@ type SolrCloudList struct {
 func init() {
 	SchemeBuilder.Register(&SolrCloud{}, &SolrCloudList{})
 }
+
+// +kubebuilder:validation:Enum=None;Want;Need
+type ClientAuthType string
+
+const (
+	None ClientAuthType = "None"
+	Want ClientAuthType = "Want"
+	Need ClientAuthType = "Need"
+)
+
+type SolrTLSOptions struct {
+	// TLS Secret containing a pkcs12 keystore
+	PKCS12Secret *corev1.SecretKeySelector `json:"pkcs12Secret"`
+
+	// Secret containing the key store password; this field is required as most JVMs do not support pkcs12 keystores without a password
+	KeyStorePasswordSecret *corev1.SecretKeySelector `json:"keyStorePasswordSecret"`
+
+	// Determines the client authentication method, either None, Want, or Need;
+	// this affects K8s ability to call liveness / readiness probes so use cautiously.
+	// +optional
```

Comment:
You can use `// +kubebuilder:default=None`, and get rid of the `withDefaults()` method.